flow/raster_cache.cc

```diff
@@ -34,16 +34,16 @@ void RasterCacheResult::draw(SkCanvas& canvas, const SkPaint* paint) const {
 
   SkRect bounds =
       RasterCacheUtil::GetDeviceBounds(logical_rect_, canvas.getTotalMatrix());
+  bounds.roundOut(&bounds);
 #ifndef NDEBUG
   // The image dimensions should always be larger than the device bounds and
   // smaller than the device bounds plus one pixel, at the same time, we must
   // introduce epsilon to solve the round-off error. The value of epsilon is
   // 1/512, which represents half of an AA sample.
   float epsilon = 1 / 512.0;
-  FML_DCHECK(image_->dimensions().width() - bounds.width() > -epsilon &&
-             image_->dimensions().height() - bounds.height() > -epsilon &&
-             image_->dimensions().width() - bounds.width() < 1 + epsilon &&
-             image_->dimensions().height() - bounds.height() < 1 + epsilon);
+  FML_DCHECK(
+      std::abs(bounds.width() - image_->dimensions().width()) <= 1 + epsilon &&
+      std::abs(bounds.height() - image_->dimensions().height()) <= 1 + epsilon);
```

Comment:
hmm, this is not good because i'm effectively reverting https://github.com/flutter/engine/pull/34660. this is tricky to get right because after rounding out, the width may be different in `RasterCache::Rasterize()` vs. `RasterCacheResult::draw()` since they can have a different translation.